postgres.go

```diff
@@ -42,7 +47,20 @@ func (dialector Dialector) Name() string {
 	return "postgres"
 }
 
-var timeZoneMatcher = regexp.MustCompile("(time_zone|TimeZone)=(.*?)($|&| )")
+func (dialector Dialector) Apply(config *gorm.Config) error {
+	var namingStartegy *schema.NamingStrategy
+	switch v := config.NamingStrategy.(type) {
+	case *schema.NamingStrategy:
+		namingStartegy = v
+	case schema.NamingStrategy:
+		namingStartegy = &v
+	case nil:
+		namingStartegy = &schema.NamingStrategy{}
+	}
+	namingStartegy.IdentifierMaxLength = defaultIdentifierLength
```

Comment:
It seems that only IdentifierMaxLength equal to 0 should be set to the default value.

Reply:
Is there a need for this specific check ??
```
if namingStartegy.IdentifierMaxLength == 0 {
		namingStartegy.IdentifierMaxLength = defaultIdentifierLength
	}
```

Isn't it obvious that it will already be zero, and does it matter if it's not 0 ??
If it's not 0 do we not want to override it to default value ( 63 ) of postgres DB ??


@a631807682

Reply:
Users can customize NamingStrategy, and all properties of NamingStrategy may be used by users, so we should not override them.
However, since IdentifierMaxLength is not a feature that was supported from the beginning, users before this feature was added may not have set it, which may become a breaking change. To avoid this problem, we can only override IdentifierMaxLength when it is 0.

Reply:
Got it !!
Applied check before assigning default value. Check again !!